src/components/smartSearch/EditTargetDialog.tsx

```diff
@@ -21,63 +21,50 @@ interface EditTargetDialogProps {
 const EditTargetDialog = ({ onDialogClose, open, filterSpec }: EditTargetDialogProps) : JSX.Element => {
     const queryClient = useQueryClient();
     const { orgId, taskId } = useRouter().query;
-    const filtersWithIds = filterSpec.map((filter, index) => ({ ...filter, id: index }));
-    const [filterArray, setFilterArray] = useState<ZetkinSmartSearchFilterWithId[]>(filtersWithIds);
-    const [selectedType, setSelectedType] = useState<FILTER_TYPE | null>(null);
-    const [selectedFilter, setSelectedFilter] = useState<ZetkinSmartSearchFilterWithId| null>(null);
+
+    const { filtersWithIds: filterArray, filters, addFilter, editFilter, deleteFilter } = useSmartSearch(filterSpec);
+    const [selectedFilter, setSelectedFilter] = useState<SelectedSmartSearchFilter>(null);
 
     const taskMutation = useMutation(patchTaskTarget(orgId as string, taskId as string),{
         onSettled: () => queryClient.invalidateQueries(['filter_spec', orgId, taskId]),
     } );
 
     const handleDialogClose = () => {
-        setSelectedType(null);
+        setSelectedFilter(null);
         onDialogClose();
     };
 
-    const handleCancelFilter = () => setSelectedType(null);
+    const handleCancelFilter = () => setSelectedFilter(null);
 
-    const handleSubmitFilter = (filter:ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter) => {
-        if (isFilterWithId(filter)) {
-            setFilterArray(filterArray.map(f => {
-                if (f.id === filter.id) {
-                    return filter;
-                }
-                else {
-                    return f;
-                }
-            }));
+    const handleSubmitFilter = (filter: ZetkinSmartSearchFilter | SmartSearchFilterWithId) => {
```

Comment:
With the change of always including ID, this could be simplified to a single `SmartSearchFilterWithId` type.